src/app/bugs/components/bugs-technology/bugs-technology.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnInit } from '@angular/core';
+import { chartsData, pieChartsData,  pieChartOPtion } from 'src/app/commons/constants/charts.constant';
+import { HttpService } from 'src/app/commons/services/http.service';
+import { SeverityModel } from '../../models/classes/severity.models.class';
+import { BugsGenericService } from '../../services/bugs-generic.service';
+import { BugsMapperService } from '../../services/bugs-mapper.service';
+
+@Component({
+  selector: 'app-bugs-technology',
+  templateUrl: './bugs-technology.component.html',
+  styleUrls: ['./bugs-technology.component.scss']
+})
+export class BugsTechnologyComponent implements OnInit {
+  public data = pieChartsData;
+  public options = pieChartOPtion;
+  public type= "pie";
+  public severityData: Array<SeverityModel>;  
+  
+  constructor(private httpService: HttpService, private bugsMapperService: BugsMapperService,
+    private bugsGenericService: BugsGenericService) { }
+
+  ngOnInit(): void {
+    this.httpService.getMethod('assets/json/prb.json').subscribe((data) => {      
+      this.severityData = this.bugsMapperService.severityMapper(data); 
```

Comment:
lets create a local variable here rather than using global properties.

something like this

const severityData: Array<SeverityModel> = this.bugsMapperService.severityMapper(data); 
this.data =  this.bugsGenericService.getTechnologyList(severityData);

Reply:
Yes removed global variable and declarer as private